programs/vote/src/vote_state/mod.rs

```diff
@@ -618,14 +618,20 @@ pub fn process_new_vote_state(
     let mut finalized_slot_count = 1_u64;
```

Comment:
Lets rename this something like `finalized_and_voted_on_count`, to indicate that we only award credits to previously voted on slots. 
Also lets initialize this with the flag here and have a comment to explain what's happening when the flag is turned off. 

```
let new_root_vote_credits_fix = feature_set... ;
let mut finalized_and_voted_on_slot_count = if new_root_vote_credits_fix { 0_u64} else {
 // Previously we were incorrectly awarding credits to new roots that were not present in the previous vote state. 
 //This is an issue because ... 
 1_u64
}

Reply:
Also update the previous comment on L612 "Start with 1 for the new root" is now outdated.

Reply:
I can do this; but keep in mind that this code is rewritten with the timely_vote_credits change and this variable will not exist after that.

Also I was trying to make the change as non-disruptive as possible (i.e. changing the fewest lines possible and not altering pre-existing logic except where necessary), but I will alter it according to these instructions.

Reply:
> Also update the previous comment on L612 "Start with 1 for the new root" is now outdated.

Also good point, in fact the comment itself gave a pretty clear description of what was wrong with the code itself!

Reply:
forgot that this code was changing with the timely vote credits as well. In that case I think it would be best to keep it under 1 feature flag as suggested below.